File: Backend/exam-server/src/main/java/com/exam/examserver/service/QuestionService.java
package com.exam.examserver.service;

import com.exam.examserver.entity.Question;
import com.exam.examserver.entity.User;

import java.util.List;
import java.util.Map;

public interface QuestionService {

    Question addQuestion(Question question);

    Question updateQuestion(Question question);

    List<Question> getAllQuestions();

    Question getByQuestionId(long quesId);

    void deleteByQuestionId(long quesId);

    List<Question> getRandomQuestionsByQuizId(long qId);

    List<Question> getAllQuestionsByQuizId(long qId);

    Map<String, Integer> evalQuiz(List<Question> questionList);
}

File: Frontend/exam-portal-fe/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private _httpClient: HttpClient) { }

  public getAllQuestionsByQuizId(qid: any) {
    return this._httpClient.get(`${baseUrl}/api/questions/quiz/all/${qid}`);
  }

  public addQuestion(question: any) {
    return this._httpClient.post(`${baseUrl}/api/questions`, question);
  }

  public getQuestionByQuestionId(quesId: any) {
    return this._httpClient.get(`${baseUrl}/api/questions/${quesId}`);
  }

  public updateQuestion(question: any) {
    return this._httpClient.put(`${baseUrl}/api/questions`, question);
  }

  public deleteQuestion(quesId: any) {
    return this._httpClient.delete(`${baseUrl}/api/questions/${quesId}`);
  }

  public getRandomQuestionsByQuizId(qid: any) {
    return this._httpClient.get(`${baseUrl}/api/questions/quiz/${qid}`);
  }

  public evalQuiz(questions: any, id: any, qid: any, cid: any) {
    return this._httpClient.post(`${baseUrl}/api/questions/eval-quiz/${id}/${qid}/${cid}`, questions);
  }
}
